Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5439,7 +5439,9 @@ void WebPageProxy::commitProvisionalPage(IPC::Connection& connection, FrameIdent
     removeAllMessageReceivers();
     RefPtr navigation = m_navigationState->navigation(provisionalPage->navigationID());
     bool didSuspendPreviousPage = navigation ? suspendCurrentPageIfPossible(*navigation, WTF::move(mainFrameInPreviousProcess), shouldDelayClosingUntilFirstLayerFlush) : false;
-    protectedLegacyMainFrameProcess()->removeWebPage(*this, m_websiteDataStore.ptr() == provisionalPage->process().websiteDataStore() ? WebProcessProxy::EndsUsingDataStore::No : WebProcessProxy::EndsUsingDataStore::Yes);
+    // Deferring shutting down process until m_legacyMainFrameProcess is updated to new process in swapToProvisionalPage.
```

Comment:
Could we create a local variable initialized from `protectedLegacyMainFrameProcess()->shutdownPreventingScope()` instead of introducing a new mechanism via an extra `CanShutDownProcess` parameter?